cli/stream_command.go

```diff
@@ -1267,16 +1269,22 @@ func (c *streamCmd) copyAndEditStream(cfg api.StreamConfig, pc *fisk.ParseContex
 		cfg.Placement = &api.Placement{}
 	}
 
-	if cfg.Placement == nil {
-		cfg.Placement = &api.Placement{}
-	}
+	// For placement constraints, we explicitly support empty strings to
+	// remove, so use the *Set bool variables to distinguish "was set on
+	// command-line" from "is not empty".
 
-	if c.placementCluster != "" {
+	if c.placementClusterSet {
 		cfg.Placement.Cluster = c.placementCluster
 	}
 
-	if len(c.placementTags) > 0 {
-		cfg.Placement.Tags = c.placementTags
+	if c.placementTagsSet {
```

Comment:
nice this turned out much better.  The `IsSetByUser()` behavior is quite new - after I wrote this code all, so nice to see it help.

Will review closely tomorrow, thank you